bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/ChannelPool.java

```diff
@@ -70,12 +73,27 @@ public synchronized void returnToPool() {
   private List<Channel> channels;
   private final AtomicInteger requestCount = new AtomicInteger();
   private final List<HeaderInterceptor> headerInterceptors;
-
-  public ChannelPool(List<Channel> channels, List<HeaderInterceptor> headerInterceptors) {
-    this.channels = new ArrayList<>(channels);
+  private ChannelFactory factory;
+  private int reservedChannelCount = 0;
+
+  public ChannelPool(List<HeaderInterceptor> headerInterceptors, ChannelFactory factory)
+      throws IOException {
+    this.channels = new ArrayList<>();
+    channels.add(factory.create());
+    this.factory = factory;
     this.headerInterceptors = headerInterceptors;
   }
 
+  public synchronized void ensureCapacity(int capacity) throws IOException {
```

Comment:
"capacity" is kind of confusing. it could means the number of channels we could create not the number of active channels.